ProjFS.Mac/Scripts/Build.sh

```diff
@@ -21,9 +21,71 @@ fi
 
 # Run Tests and put output into a xml file
 set -o pipefail
-xcodebuild -configuration $CONFIGURATION -project $PROJFS/PrjFS.xcodeproj -scheme 'Build All' test | xcpretty -r junit --output $PROJFS/TestResultJunit.xml || exit 1
+xcodebuild -configuration $CONFIGURATION -enableCodeCoverage YES -project $PROJFS/PrjFS.xcodeproj -derivedDataPath $ROOTDIR/BuildOutput/ProjFS.Mac/Coverage -scheme 'Build All' test | xcpretty -r junit --output $PROJFS/TestResultJunit.xml || exit 1
 set +o pipefail
 
+shopt -s nullglob
+COVERAGE_FILES=($ROOTDIR/BuildOutput/ProjFS.Mac/Coverage/Logs/Test/*.xcresult/*/*xccovreport)
+shopt -u nullglob
+COVERAGE_FILE_COUNT="${#COVERAGE_FILES[@]}"
+if (( COVERAGE_FILE_COUNT == 0 )); then
+  echo "Error: No coverage files found"
+  exit 1
+fi
+
+if !(gem list --local | grep xcperfect); then
+  echo "Attempting to run 'sudo gem install xcperfect'.  This may ask you for your password to gain admin privileges"
+  sudo gem install xcperfect
+fi
+
+xcrun xccov view "${COVERAGE_FILES[(COVERAGE_FILE_COUNT - 1)]}" --json | xcperfect 
+
+# Example the latest code coverage file
+xcrun xccov view "${COVERAGE_FILES[(COVERAGE_FILE_COUNT - 1)]}" > $PROJFS/CoverageResult.txt 
+
+# Fail on any line that doesn't show %100 coverage and isn't on the exclusion list or hpp/cpp
+while read line; do
+  if [[ $line != *"100.00%"* ]] && 
```

Comment:
We need all these exclusions since we're including too many files in the test report.  

@pmj do you know how we could switch the code coverage report to just be on KauthHandler.cpp && VirtualizationRoots.cpp.

Alternatively we could parse the json and remove the files we don't care about in that report.  @mjcheetham know a fancy json parser for MacOS as a backup?

Reply:
You could use `jq` which is very small, lightweight and installable by Homebrew.. it's like "sed/awk for JSON"

Reply:
Thanks Matt!  I had to turn off json since the build output didn't support the output from tool.  Once this is supported I think the aim should be to move json full stop and jq looks like a great tool for that.

Reply:
> Alternatively we could parse the json

@jeschu1 for my own understanding, which of the above commands is producing the json?

Reply:
@wilbaker the "--json" option does it

`#xcrun xccov view "$COVERAGE_FILE" --json `